### PR TITLE
Semantic creation functions

### PR DESCRIPTION
**Pull request by Meissnerinc**
Create semantics function now only returns the semantics and does not alter the original dataframe

### CODE REVIEW

**Review by bosyk — CHANGES_REQUESTED** [review]
The test `test_upload_event_log_from_df` fails at line 92 and the following:

```python
resp = self.api.upload_event_log_df(
    'pylana-test-log-from-df', df_log, df_case, time_format='yyyy-MM-dd HH:mm:ss')

self.assertEqual(resp.status_code, 200)
```

which means the method `upload_event_log_df` is likely broken, assumed that the test expectation hasn't changed.

File: pylana/utils.py [hunk 1]
```diff
@@ -62,40 +62,36 @@ def create_event_semantics_from_df(df: pd.DataFrame, time_format: str = "yyyy-MM
...
-        for col in df.select_dtypes('number').columns
-        if col not in bares
-    ]
-    dct_bare = [{**{'idx': n}, **dct} for n, dct in enumerate(dct_bare)]
-
-    return df.loc[:, [c['name'] for c in dct_bare]], dct_bare
+    bares = ['Case ID', 'Action']
+    dct_bare = []
+    for i in range(0, len(df.columns)):
+        col = df.columns[i]
```
Comment (bosyk) [inline pylana/utils.py]: This can be pythonically simplified to

```python
for i, col in enumerate(df.columns):
```

File: pylana/utils.py [hunk 2]
```diff
@@ -62,40 +62,36 @@ def create_event_semantics_from_df(df: pd.DataFrame, time_format: str = "yyyy-MM
...
+                'name': col,
+                'semantic': col,
+                'format': time_format,
+                'idx': i}]
+        else:
+            dct_bare += [{
+                'name': col,
+                'semantic': col,
+                'format': None,
+                'idx': i}]
```
Comment (bosyk) [inline pylana/utils.py]: Personally, I dislike `if`s with several branches. It always looks much more complex than the task is. Nevertheless, we can make it slightly more readable:

```python
ids = ["Case ID", "Action"]
timestamps = ["Start", "Complete"]

if col in ids:
    # the most general branch, that exists in all logs
    dct_bare += [{
        'name': col,
        'semantic': col,
        'format': None,
        'idx': i}]
elif col in timestamps:
    # the second most general branch, Start exists always, Complete sometimes
    dct_bare += [{
        'name': col,
        'semantic': col,
        'format': time_format,
        'idx': i}]
else:
    # an optional branch, defined by not being (partially) required and thus inferred
    if pd.api.types.is_object_dtype(df[col]):
        dct_bare += [{
            'name': col,
            'semantic': 'CategorialAttribute',
            'format': None,
            'idx': i}]
    elif pd.api.types.is_numeric_dtype(df[col]):
        dct_bare += [{
            'name': col,
            'semantic': 'NumericAttribute',
            'format': None,
            'idx': i}]
```

If we expect to have most of the times less than a couple thousand columns, we can use a comprehension like

```python
bares = ['Case ID', 'Action']
dct_bares = [{
    'name': col,
    'semantic': col,
    'format': None,
    'idx': i}
    for i, col in enumerate(df.columns)
    if col in bares
]
```

for all branches and then combine them into the result.

File: pylana/utils.py (hunk 1 above)
Comment (Meissnerinc) [reply pylana/utils.py]: done

File: pylana/utils.py (hunk 2 above)
Comment (Meissnerinc) [reply pylana/utils.py]: done

File: pylana/utils.py [hunk 3]
```diff
@@ -62,40 +62,42 @@ def create_event_semantics_from_df(df: pd.DataFrame, time_format: str = "yyyy-MM
...
-        'semantic': 'NumericAttribute',
-        'format': None}
-        for col in df.select_dtypes('number').columns
-        if col not in bares
-    ]
-    dct_bare = [{**{'idx': n}, **dct} for n, dct in enumerate(dct_bare)]
-
-    return df.loc[:, [c['name'] for c in dct_bare]], dct_bare
+    dct_bare = []
+    ids = ["Case ID", "Action"]
```
Comment (bosyk) [inline pylana/utils.py]: Space character are not nice, since it prevents accessing a data frame atribute lime `df.name`. Better use `CaseID`.

Comment (bosyk) [reply pylana/utils.py]: I realised that the issue with the case id naming was purely a lana issue, so I introduced a mapping between a required name for the case id column in python. Thus, we can allow case id names we like, currently `Case_ID` and `CaseID`.

**Review by bosyk — APPROVED** [review]
I made the case id column naming independent from the lana api convention and made the code slightly more pythonic by removing reliance on indices.

**Pull request merged**